dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/TrackerPreheat.java

```diff
@@ -460,29 +465,64 @@ public void putNotes( List<TrackedEntityComment> trackedEntityComments )
         return Optional.ofNullable( notes.getOrDefault( TrackerIdScheme.UID, new HashMap<>() ).get( uid ) );
     }
 
-    public Relationship getRelationship( TrackerIdScheme identifier, String relationship )
+    public Relationship getRelationship( TrackerIdScheme identifier,
+        org.hisp.dhis.tracker.domain.Relationship relationship )
     {
         if ( !relationships.containsKey( identifier ) )
         {
             return null;
         }
 
-        return relationships.get( identifier ).get( relationship );
+        Optional<RelationshipType> relationshipTypeFromPreheated = relationships.get( identifier ).values()
```

Comment:
I think here the logic could be simplified.
In the preheat we are going to have all the relationshipTypes.
So we can just check like this:
get( RelationshipType.class,  relationship.getRelationshipType() )

Reply:
yes you are right, I'll change this